server/cloud_limits.go

```diff
@@ -0,0 +1,71 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/mattermost/mattermost-server/v6/model"
+	"net/http"
+)
+
+const cloudMaxParticipants = 8
+
+// JoinAllowed returns true if the user is allowed to join the call, taking into
+// account cloud limits
+func (p *Plugin) joinAllowed(channelID string, state *channelState) (bool, error) {
+	license := p.pluginAPI.System.GetLicense()
+
+	// Rules are:
+	// On-prem: no limits to calls
+	// Cloud Starter: DMs 1-1 only
+	// Cloud Professional & Cloud Enterprise: DMs 1-1, GMs and Channel calls limited to 8 people.
+
+	if !isCloud(license) {
+		return true, nil
+	}
+
+	channel, err := p.API.GetChannel(channelID)
+	if err != nil {
+		return false, fmt.Errorf("get channel failed: %w", err)
+	}
```

Comment:
Since we are fetching the channel already in `handleJoin` maybe we can pass it through `addUserSession` in place of  the channel id and avoid some redundancy.